_analyzers/index.md

```diff
@@ -15,16 +15,24 @@ redirect_from:
 
-Text analysis consists of the following steps:
+The objective of text analysis is to split the unstructured free-text content of the source document into a sequence of terms, which are then stored in an inverted index. Subsequently, when a similar text analysis is applied to a user's query, the resulting sequence of terms facilitates the matching of relevant source documents.
 
-1. _Tokenize_ text into terms: For example, after tokenization, the phrase `Actions speak louder than words` is split into tokens `Actions`, `speak`, `louder`, `than`, and `words`.
-1. _Normalize_ the terms by converting them into a standard format, for example, converting them to lowercase or performing stemming (reducing the word to its root): For example, after normalization, `Actions` becomes `action`, `louder` becomes `loud`, and `words` becomes `word`.
+From a technical point of view, the text analysis process consists of several steps, some of which are optional:
+
+1. Before the free-text content can be split into individual words, it may be beneficial to refine the text at a character level. The primary aim of this optional step is to help the tokenizer (the subsequent stage in the analysis process) generate better tokens. This can include removal of markup tags (such as HTML) or handling a specific character patterns (like replacing the &#x1F642; emoji with the text `:slightly_smiling_face:`).
+
+2. The next step is to split the free-text into individual words---_tokens_.. This is a job of a _tokenizer_. For example, after tokenization, the sentence `Actions speak louder than words` is split into tokens `Actions`, `speak`, `louder`, `than`, and `words`.
+
+3. The last step is to process individual tokens by applying a series of token filters. The aim is to convert each token into a predictable form that is directly stored in the index, for example, by converting them to lowercase or performing stemming (reducing the word to its root). For example, the token `Actions` becomes `action`, `louder` becomes `loud`, and `words` becomes `word`.
+
+Although the terms ***token*** and ***term*** may sound similar and are occasionally used interchangeably, it is helpful to understand the difference between the two. In the context of Apache Lucene®, each holds a distinct role. A ***token*** is created by a tokenizer during text analysis and often undergoes a number of additional modifications as it passes through the chain of token filters. Each token is associated with metadata that can be further used during the text analysis process. A ***term*** is a data value that is directly stored in the inverted index and is associated with a lot less metadata. During search, matching operates at the term level.
```

Comment:
@natebower I am checking Apache guide on project name usage https://www.apache.org/foundation/marks/guide#naming and it seems that dropping "(R)" should be fine if:

> The first and most prominent mentions must use the full form: "Apache Hadoop®" of the name for any individual usage (webpage, handout, slides, etc.) Depending on the context and writing style, you should use the full form of the name sufficiently often to ensure that readers clearly understand the association of both the Hadoop project and the Hadoop software product to the ASF as the parent organization.
>
> Later in each specific document you may use the bare form of the name, i.e. just Hadoop, as best suits your writing style.

It seems legit to drop the (R) here. 👍

Reply:
@lukas-vlcek The ® is mostly for Apache to provide notice to the public that a trademark or service mark is registered. Although the symbols ® and ™ (for registered and unregistered trademarks, respectively) often accompany trademark names on product packaging and in promotional material, there is no legal requirement to use these symbols, and they should be omitted wherever possible.

Reply:
@natebower Sure, but still you do not want to give anybody an opportunity to publicly point at you or your employer for not using legal attributions correctly (unfortunately this can happen).